File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.c
#include <stdio.h>
#include <stdlib.h>

int main(void) { 
   int mushroom[10], prev_sum, result;
   for(int i = 0; i < 10; i++){
      scanf("%d", &mushroom[i]);
   }

   int sum = 0;
   for(int i = 0; i < 10; i++){
      prev_sum = sum;
      
      if(sum + mushroom[i] <= 100){
         sum += mushroom[i];
      }
      else{
         sum += mushroom[i];
         if(abs(prev_sum - 100) < abs(sum +  - 100)){
            result = prev_sum;
            printf("%d", result);

            return 0;
         }
         else{
            result = sum;
            printf("%d", result);

            return 0;
         }
      }
   }

   
   printf("%d", sum);
   return 0;
}

File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.py
mushroom = []
for _ in range(10):
  elem = int(input())
  mushroom.append(elem)

eat = 0
sum, result = 0, 0
for item in mushroom:
  prev = sum
  sum += item
  
  if sum >= 100:
    if(abs(prev - 100) < abs(sum - 100)):
      result = prev
    else:
      result = sum

    print(result)
    break

  eat += 1

if(eat == 10):
  print(sum)